### ISSUE TITLE
Confusing error message for //visibility:none

### ISSUE THREAD

**Issue by aiuto** [bazelbuild/bazel#22279]
### Description of the bug:

If you misspell `//visibility:private` (or `public`), you get a misleading error message. It talks about the package, rather than the special rule class.

```
.../BUILD:2:8: no such package 'visibility': BUILD file not found in any of the following directories. Add a BUILD file to a directory to mark it as a package.
```

The message could be  "The value must be //visbility:public or //visibiility:private'

### Which category does this issue belong to?

Core

### What's the simplest, easiest way to reproduce this bug? Please provide a minimal example if possible.

```
genrule(
  name = "x",
  cmd = "echo x >$@",
  outs = ["x.txt"],
  visibility = ["//visibility:none"],
)
```

### What is the output of `bazel info release`?

head

**Comment by tetromino** [comment]
@bazel-io fork 7.2.0

**Comment by iancha1992** [comment]
A fix for this issue has been included in [Bazel 7.2.0 RC1](https://github.com/bazelbuild/bazel/releases/tag/7.2.0rc1). Please test out the release candidate and report any issues as soon as possible.
If you're using Bazelisk, you can point to the latest RC by setting USE_BAZEL_VERSION=7.2.0rc1. Thanks!
